client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	getShortCodeAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+		const defaultWidth = 640;
+		const defaultHeight = defaultWidth * 9 / 16;
+		return {
```

Comment:
Using `Object.assign` (or similarly, [`_.assign`](https://lodash.com/docs#assign) or [`_.defaults`](https://lodash.com/docs#defaults)) can be a handy way to clearly identify the default values, while still allowing override if value is known:

```js
return Object.assign( {
	w: defaultWidth,
	h: defaultHeight,
	at: 0,
	defaultLangCode: false
}, {
	videopress_guid: shortcode.attrs.numeric[0],
	w: parseInt( namedAttrs.w, 10 ),
	h: parseInt( namedAttrs.h, 10 ),
	autoplay: namedAttrs.autoplay === 'true',
	hd: namedAttrs.hd === 'true',
	loop: namedAttrs.loop === 'true',
	at: parseInt( namedAttrs.at, 10 ),
	defaultLangCode: namedAttrs.defaultlangcode
} );
```

Might be a negligible performance difference, though the clarity benefits should outweigh this. What do you think?